frontend/lib/framework/error-classes-list.cpp

```diff
@@ -838,6 +838,42 @@ void ErrorValueUsedAsType::write(ErrorWriterBase& wr) const {
   // wr.message("Did you mean to use '.type'?");
 }
 
+void ErrorDeprecation::write(ErrorWriterBase& wr) const {
+  auto msg = std::get<std::string>(info);
+  auto mention = std::get<const uast::AstNode*>(info);
+  auto target = std::get<const uast::NamedDecl*>(info);
+  CHPL_ASSERT(mention && target);
+
+  wr.heading(kind_, type_, mention, msg);
+  wr.code(mention, {mention});
+
+  /* TODO: Need to know whether the symbol is user or not:
```

Comment:
Just to double check, the idea here is to only print out the `declared here:` for user code?

Reply:
Yeah.

Reply:
We have `idIsInBundledModule` and `idIsInInternalModule` if you want to do something about it soon.

Reply:
I'm probably going to use those in conjunction to write a helper function `isAstUserCode` or something.